cpp/models/ide_secir/model.cpp

```diff
@@ -195,49 +250,93 @@ void Model::compute_flow(int idx_InfectionTransitions, Eigen::Index idx_Incoming
     m_transitions.get_last_value()[Eigen::Index(InfectionTransition::SusceptibleToExposed)] =
         dt * m_forceofinfection * m_populations.get_last_value()[Eigen::Index(InfectionState::Susceptible)];
+
     // Calculate all other flows with compute_flow.
-    // Flow from E to C
+    // Flow ExposedToInfectedNoSymptoms.
     compute_flow((int)InfectionTransition::ExposedToInfectedNoSymptoms,
                  Eigen::Index(InfectionTransition::SusceptibleToExposed), dt);
-    // Flow from C to I
+    // Flow InfectedNoSymptomsToInfectedSymptoms.
     compute_flow((int)InfectionTransition::InfectedNoSymptomsToInfectedSymptoms,
                  Eigen::Index(InfectionTransition::ExposedToInfectedNoSymptoms), dt);
-    // Flow from C to R
+    // Flow InfectedNoSymptomsToRecovered.
     compute_flow((int)InfectionTransition::InfectedNoSymptomsToRecovered,
                  Eigen::Index(InfectionTransition::ExposedToInfectedNoSymptoms), dt);
-    // Flow from I to H
+    // Flow InfectedSymptomsToInfectedSevere.
     compute_flow((int)InfectionTransition::InfectedSymptomsToInfectedSevere,
                  Eigen::Index(InfectionTransition::InfectedNoSymptomsToInfectedSymptoms), dt);
-    // Flow from I to R
+    // Flow InfectedSymptomsToRecovered.
     compute_flow((int)InfectionTransition::InfectedSymptomsToRecovered,
                  Eigen::Index(InfectionTransition::InfectedNoSymptomsToInfectedSymptoms), dt);
-    // Flow from H to U
+    // Flow InfectedSevereToInfectedCritical.
     compute_flow((int)InfectionTransition::InfectedSevereToInfectedCritical,
                  Eigen::Index(InfectionTransition::InfectedSymptomsToInfectedSevere), dt);
-    // Flow from to H to R
+    // Flow InfectedSevereToRecovered.
     compute_flow((int)InfectionTransition::InfectedSevereToRecovered,
                  Eigen::Index(InfectionTransition::InfectedSymptomsToInfectedSevere), dt);
-    // Flow from U to D
+    // Flow InfectedCriticalToDead.
     compute_flow((int)InfectionTransition::InfectedCriticalToDead,
                  Eigen::Index(InfectionTransition::InfectedSevereToInfectedCritical), dt);
-    // Flow from U to R
+    // Flow InfectedCriticalToRecovered.
     compute_flow((int)InfectionTransition::InfectedCriticalToRecovered,
                  Eigen::Index(InfectionTransition::InfectedSevereToInfectedCritical), dt);
 }
 
-void Model::compute_deaths()
+void Model::update_compartments_current_timestep()
 {
     Eigen::Index num_time_points = m_populations.get_num_time_points();
 
+    // Exposed
+    m_populations.get_last_value()[Eigen::Index(InfectionState::Exposed)] =
+        m_populations[num_time_points - 2][Eigen::Index(InfectionState::Exposed)] +
+        m_transitions.get_last_value()[Eigen::Index(InfectionTransition::SusceptibleToExposed)] -
+        m_transitions.get_last_value()[Eigen::Index(InfectionTransition::ExposedToInfectedNoSymptoms)];
```

Comment:
Quick thought: Wouldn't it be more concise to have a function `update_compartment_from_flow(Eigen::Index idx_InfectionState, std::vector<Eigen::Index> idx_IncomingFlows, std::vector<Eigen::Index> idx_OutgoingFlows)` ?

Reply:
In principle, we could simply implement it, but then we would have one more function, making it more complex. Would you like to have such a function?

Reply:
Not sure, just thought about simpler extension to new models (maybe make it less error prone with the +/- signs?) 
Maybe you can have a 5min discussion with @reneSchm and then you decide ;)